src/legacy/core_plugins/region_map/public/util.ts

```diff
@@ -17,14 +17,12 @@
  * under the License.
  */
 
-import _ from 'lodash';
-
+import { FileLayer, VectorLayer } from 'ui/vis/map/service_settings';
 // TODO: reference to TILE_MAP plugin should be removed
 import { ORIGIN } from '../../../../legacy/core_plugins/tile_map/common/origin';
 
-export function mapToLayerWithId(prefix, layer) {
-  const clonedLayer = _.cloneDeep(layer);
-  clonedLayer.layerId = prefix + '.' + layer.name;
-  clonedLayer.isEMS = ORIGIN.EMS === prefix ?  true : false;
-  return clonedLayer;
-}
+export const mapToLayerWithId = (prefix: string, layer: FileLayer): VectorLayer => ({
+  ...layer,
```

Comment:
The removal of cloneDeep means that the `FileLayerField` objects and the array of those objects in the layer will reference the original ones. Are we sure they are not mutated somewhere? This could cause some bugs that are hard to track down.

It might be totally fine, just to make sure as I'm not familiar with the control flow in that part of the code.

Reply:
I've checked the code flow and these values are read-only. Also `FileLayer` array comes straight from the server, so they are safe for modifying.
Of course I could keep the `cloneDeep` here, but didn't see a necessity in it.